rectools/models/sasrec.py

```diff
@@ -52,6 +60,119 @@ def forward(self, seqs: torch.Tensor, timeline_mask: torch.Tensor, attn_mask: to
         raise NotImplementedError()
 
 
+class CatFeatureEmebbedingsItem(nn.Module):
+    """Сlass for single category feature embeddings. TODO"""
+
+    def __init__(self, n_factors: int, n_values: int, dropout_rate: float):
+        super().__init__()
+
+        self.n_values = n_values
+        self.value_emb = nn.Embedding(num_embeddings=n_values, embedding_dim=n_factors)
+        self.drop_layer = nn.Dropout(dropout_rate)
+
+    def forward(self, values: torch.Tensor) -> torch.Tensor:
+        """TODO"""
+        feature_embeddings = self.value_emb(values)
+        feature_embeddings = self.drop_layer(feature_embeddings)
+        return feature_embeddings
+
+    @property
+    def device(self) -> torch.device:
+        """TODO"""
+        return self.value_emb.weight.device
+
+    @property
+    def catalogue_values(self) -> torch.Tensor:
```

Comment:
This property will only be necessary for `ItemNetConstructor` class. Jther item net blocks do not need it at all.
Same for `get_all_embeddings`.

This comes because of our decision to accept `item_net_blocks: tp.List[tp.Type[ItemNetBlockBase]]` instead of `item_net_type: tp.Type[ItemNetBase]` in model interface.